src/darwin/CHIPTool/CHIPTool/View Controllers/QRCode/QRCodeViewController.m

```diff
@@ -628,76 +628,42 @@ - (void)retrieveAndSendWiFiCredentials
     [self presentViewController:alertController animated:YES completion:nil];
 }
 
-- (void)addOrUpdateWiFiNetwork:(NSString *)ssid password:(NSString *)password
+- (void)commissionWithSSID:(NSString *)ssid password:(NSString *)password
 {
-    CHIPDevice * chipDevice = CHIPGetDeviceBeingCommissioned();
-    if (chipDevice) {
-        self.cluster = [[CHIPNetworkCommissioning alloc] initWithDevice:chipDevice endpoint:0 queue:dispatch_get_main_queue()];
-        __auto_type * params = [[CHIPNetworkCommissioningClusterAddOrUpdateWiFiNetworkParams alloc] init];
-        params.ssid = [ssid dataUsingEncoding:NSUTF8StringEncoding];
-        params.credentials = [password dataUsingEncoding:NSUTF8StringEncoding];
-        params.breadcrumb = @(0);
-
-        __weak typeof(self) weakSelf = self;
-        [self->_cluster
-            addOrUpdateWiFiNetworkWithParams:params
-                           completionHandler:^(CHIPNetworkCommissioningClusterNetworkConfigResponseParams * _Nullable response,
-                               NSError * _Nullable error) {
-                               // TODO: addWiFiNetworkWithParams
-                               // returns status in its response,
-                               // not via the NSError!
-                               [weakSelf onAddNetworkResponse:error isWiFi:YES];
-                           }];
-    } else {
-        NSLog(@"Status: Failed to find a device being commissioned");
-    }
-}
 
-- (void)addOrUpdateThreadNetwork:(NSData *)threadDataSet
-{
-    CHIPDevice * chipDevice = CHIPGetDeviceBeingCommissioned();
-    if (chipDevice) {
-        self.cluster = [[CHIPNetworkCommissioning alloc] initWithDevice:chipDevice endpoint:0 queue:dispatch_get_main_queue()];
-        __auto_type * params = [[CHIPNetworkCommissioningClusterAddOrUpdateThreadNetworkParams alloc] init];
-        params.operationalDataset = threadDataSet;
-        params.breadcrumb = @(0);
-
-        __weak typeof(self) weakSelf = self;
-        [self->_cluster
-            addOrUpdateThreadNetworkWithParams:params
-                             completionHandler:^(CHIPNetworkCommissioningClusterNetworkConfigResponseParams * _Nullable response,
-                                 NSError * _Nullable error) {
-                                 // TODO: addThreadNetworkWithParams
-                                 // returns status in its response,
-                                 // not via the NSError!
-                                 [weakSelf onAddNetworkResponse:error isWiFi:NO];
-                             }];
-    } else {
-        NSLog(@"Status: Failed to find a device being  commissioned");
+    NSError *error;
+    CHIPDeviceController * controller = [CHIPDeviceController sharedController];
+    // create commissioning params in ObjC. Pass those in here with network credentials.
+    // maybe this just becomes the new norm
+    CHIPCommissioningParameters * params = [[CHIPCommissioningParameters alloc] init];
```

Comment:
```suggestion
    CHIPCommissioningParameters * params = [CHIPCommissioningParameters new];
```
is more modern usage, I thought?  But if not, leave as is.

Reply:
They're almost the same, and alloc + init fits better with other init APIs where init is overridden (for example `alloc] initWithParams:]`).